commit cd98eb5dbfbac18226bb7e5cc07465bc484d7552 (refs/changes/33/4433/3)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2018-06-28T08:22:58+01:00 (1 year, 3 months ago)
    
    TRITON-551 sdcadm build fails as non-root, trying to create log dir
    TRITON-552 components using mk-shar during their build fail in /usr/bin/cp

diff --git a/lib/logging.js b/lib/logging.js
index dddbd42..72ebdd6 100644
--- a/lib/logging.js
+++ b/lib/logging.js
@@ -196,6 +196,7 @@ function createLogger(opts) {
             level: 'trace',
             path: logFile
         });
+	mkdirp.sync(LOG_DIR);
     } else {
         logStreams.push({
             type: 'raw',
@@ -207,8 +208,6 @@ function createLogger(opts) {
         });
     }
 
-    mkdirp.sync(LOG_DIR);
-
     return bunyan.createLogger({
         name: opts.name,
         component: opts.component,
diff --git a/tools/mk-shar b/tools/mk-shar
index 97fa44e..5e92113 100755
--- a/tools/mk-shar
+++ b/tools/mk-shar
@@ -88,7 +88,7 @@ cp -PR \
     $TOP/build/node \
     $PROTO/node
 mkdir -p $PROTO/tools
-cp -P $TOP/tools/rotate-logs.sh $PROTO/tools/
+cp $TOP/tools/rotate-logs.sh $PROTO/tools/
 
 # Trim out cruft for smaller package.
 rm -rf $PROTO/node/bin/npm
